sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -66,7 +66,10 @@ def get_responses():
     for work_request in control_stub.Control(get_responses()):
       logging.info('Got work %s', work_request.instruction_id)
       request_type = work_request.WhichOneof('request')
-      if request_type == ['process_bundle_progress']:
+      # WhichOneOf returns the name of the set field as a single string
+      # FIXME: angoenka Race condition can happen between process_bundle request and process_bundle_progress request
```

Comment:
Not sure what this fixme is intending to address, the separateness of threads is intentional, and the interaction happens via (explicitly documented) progress trackers.

Reply:
Fixme Explanation:
This fix will have following repercussion and I wanted to make sure that its in line with API contract.

Now that process _bundle_progress will run on different thread, its possible that process_bundle_progress and process_bundle request will be executed in out of order.
This might result into case where worker has accepted the process_bundle request but still it might throw exception on process_bundle_progress.